aiosmtpd/controller.py

```diff
@@ -17,6 +20,24 @@
 AsyncServer = asyncio.base_events.Server
 
 
+def get_localhost() -> str:
+    try:
+        sock = create_connection(("::1", 0))
+        # It should be impossible... but it worked!
+        # So we need to close it
+        sock.close()
+        # And since it worked, it's clear that IPv6 is supported
+        return "::1"
+    except OSError as e:
+        if e.errno == errno.EADDRNOTAVAIL or e.errno == 10049:
+            # 10049 is WSAEADDRNOTAVAIL .. seems to only exist on Windows
+            # So we'll not use an errno constant there.
+            return "127.0.0.1"
+        if e.errno == errno.ECONNREFUSED:
+            return "::1"
+        raise
```

Comment:
Detect "::1" by trying to open a connection to it. If successful, return "::1". If not, let's analyze the exception and respond accordingly.